app/models/container_association.rb

```diff
@@ -1,18 +1,6 @@
 # This class allows plates to be associated with wells. It was initially more flexible
 # but that flexibility has not been used and has resulted in complications elsewhere.
 class ContainerAssociation < ApplicationRecord
-  # Rails doesn't handle through associations very well when single table inheritance is also in play.
-  # For example the following:
-  # Plate.first.wells.includes(:plate)
-  # generates invalid sql, where it attempts to eager_load the container associations. It tries to filter
-  # on assets.sti_type, but doesn't join on the assets table.
-  # Here we force the join, and ensure everything works as intended. It is also possible to switch to
-  # specifying the class_name: 'Asset' on each association, but this has the side effect of disrupting
-  # other eager loading as associations aren't specified on Asset.
-  # Ideally we should be avoiding using single table inheritance in this manner; this is far from the only
-  # place it has caused pain.
-  default_scope ->() { includes(:plate, :well) }
-
```

Comment:
This has been removed as I've moved the same scope onto the relationships in plate and well. This ensures we only load the relationship appropriate to the direction were traversing in. (Ie. wells only load the plate when finding the plate, rather than also re-loading themselves)